src/substrate/filters/enricher.ts

```diff
@@ -101,69 +108,95 @@ export async function Enrich(
-          const key = accountId.toString();
-          validatorInfo[key] = {}
+        for (let validator of validators) {
+          const key = validator.toString();
+
+          const preference = api.query.staking.erasValidatorPrefs
+            // for version >= 38
+            ? await api.query.staking.erasValidatorPrefs.at(hash, currentEra, key)
+            // for version == 31
+            : await api.query.staking.validators.at(hash, key);
+
+          const commissionPer = (+preference.commission || 0) / 10_000_000;
+
+          const rewardDestination = await api.query.staking.payee.at(hash, key);
+          const controllerId = await api.query.staking.bonded.at(hash, key);
+
           validatorInfo[key] = {
             commissionPer,
-            controllerId,
-            rewardDestination,
-            nextSessionIds: await (await api.query.session.nextKeys(key)).toString(),
-            eraPoints: Number(eraPointsIndividual[key])
+            controllerId: controllerId.isSome ? controllerId.toString() : key,
+            rewardDestination: rewardDestination,
+            eraPoints: validatorEraPoints[key] ? validatorEraPoints[key] : 0
           };
-        });
-        // erasStakers(EraIndex, AccountId): Exposure -> api.query.staking.erasStakers // KUSAMA
-        // stakers(AccountId): Exposure -> api.query.staking.stakers // EDGEWARE
-        const stakersCall = (api.query.staking.stakers)
-          ? api.query.staking.stakers
-          : api.query.staking.erasStakers;
-        const stakersCallArgs = (account) => (api.query.staking.stakers)
-          ? [account]
-          : [currentEra as unknown as EraIndex, account];
-        let activeExposures: { [key: string]: any } = {}
+        };
+
+        const stakersCall = (account): Promise<Exposure> => {
+          return api.query.staking.erasStakers
+            ? api.query.staking.erasStakers.at(hash, currentEra, account)
+            : api.query.staking.stakers.at(hash, account)
+        }
+
+        let activeExposures: ActiveExposure = {};
         if (validators && currentEra) { // if currentEra isn't empty
-          active = validators.validators;
-          waiting = validators.nextElected;
-          await Promise.all(active.map(async (validator) => {
-            const tmp_exposure = (await stakersCall(...stakersCallArgs(validator))) as Exposure;
-            const exposure = {
-              own: tmp_exposure.own,
-              total: tmp_exposure.total,
-              others: tmp_exposure.others
-            }
-            activeExposures[validator.toString()] = exposure;
+          await Promise.all(validators.map(async (validator) => {
+            const tmpExposure = await stakersCall(validator);
+            activeExposures[validator.toString()] = {
+              own: +tmpExposure.own,
+              total: +tmpExposure.total,
+              others: tmpExposure.others
```

Comment:
This "other" value is still using polkadot Codec types. This needs to be changed in `types.ts`.